File: php/features.php
<?php
declare(strict_types=1);

// WorldTime SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class WorldTimeFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new WorldTimeBaseFeature();
            case "test":
                return new WorldTimeTestFeature();
            default:
                return new WorldTimeBaseFeature();
        }
    }
}

File: py/features.py
# WorldTime SDK feature factory

from feature.base_feature import WorldTimeBaseFeature
from feature.test_feature import WorldTimeTestFeature


def _make_feature(name):
    features = {
        "base": lambda: WorldTimeBaseFeature(),
        "test": lambda: WorldTimeTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
